opentracing-api/src/main/java/io/opentracing/ScopeManager.java

```diff
@@ -91,6 +91,16 @@
      */
     Span activeSpan();
 
+    /**
+     * Clear the current active state, setting both {@link #active()} and
+     * {@link #activeSpan()} to null for the current context (usually a thread).
+     *
+     * <p>
+     * This method can be called to discard any previously leaked {@link Scope} objects
+     * that were unintentionally left active.
+     */
+    void clear();
```

Comment:
Is there any difference in expected behavior between this calling 
```java
activate(null)
```

Reply:
Some implementations may keep a stack of activated scopes. My understanding is that `clear()` would clear the whole stack and `activate` would push another element on the stack.

Not quite sure but `clear()` may have slightly different semantics depending on whether the ScopeManager maintains a stack of Scopes or only holds the currently active scope.

Reply:
:+1+ My context manager -by default- maintains a stack.

Reply:
Sorry for the OT, but I'm thinking about doing the same. What are your reasons for that? Is your context manager on GitHub?

Reply:
In our specific implementation, we have to get ahold of the bottom of the stack aka. entry span. Also, the scopes don't need to reference the previously active span. This simplifies scopes and reduces GC overhead a little, as the scope object does not have to have to store a reference to the active span in a field.

Reply:
_Since you ask_: My context manager is 'pluggable' and has no real opinion on stacking or not. The default context uses a stack so you can easily change e.g. the current `Locale` just in a small piece of code and have everything else the way it was. Current supporte contexts include `MDC`, Spring security context, `Locale`, originating `ServletRequest`, _and_ OpenTracing `Span`. You can plug additional types via a ServiceLoader SPI.
For many situations, having a Closeable context that can be nested is the 'easiest' for users (as it returns things to the way they were). For OpenTracing `Spans` however, I just delegate to the configured `ScopeManager`.  
_More info_: https://github.com/talsma-ict/context-propagation  
For questions, feel free to ping me at gitter, always curious what others think.

Reply:
Tradeoffs for stacking: 
+ : Support nesting
+ : User experience: 'return things they way they were' / 'works as expected' in many cases
- : Obviously a bigger chance of memory leaks if open-close semantics are not correctly used.
- : More memory consumption / garbage collecting

Reply:
> More memory consumption / garbage collecting

Not sure if that is true. The stack won't be collected unless the thread is collected.

Reply:
I don't think this is a good idea. It is a hack, but a similar hack can be achieved by allowing a closure that clears a scope. ex passing a null tracecontext to the thing that scopes the active span

Reply:
the benefit of passing null is you can still provide for "descoping that null" which is in easiest form returning a noop scope object when it was already null.